spec/integration/post_index_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'User post index page', type: :system do
+  let!(:user) do
+    User.create(name: 'John', photo: 'https://example.com/profile.jpg')
+    User.create(name: 'Daniel', photo: 'https://example.com/profile.jpg')
+  end
+
+  let!(:posts) do
+    user.posts.create([
+                        { text: 'First post', comments_counter: 2, likes_counter: 5 },
+                        { text: 'Second post', comments_counter: 3, likes_counter: 7 },
+                        { text: 'Third post', comments_counter: 8, likes_counter: 17 }
+                      ])
+  end
+
+  it 'displays user information and posts' do
+    visit user_posts_path(user)
+
+    expect(page).to have_button('Pagination')
+  end
+
+  it 'redirects to post show page when a post is clicked' do
+    visit user_posts_path(user)
+    expect(page).to have_css('.post-show')
+    expect(page).to have_content('John')
```

Comment:
- [ ] This test scenario fails whenever I run it. It expects to see the text _**John**_ in the `Post#Show` page for **_Daniel's_** profile when Daniel's profile is clicked on the `Post#Index` page. When you created the users in the `let` block, the last profile that will be returned is that of **_Daniel_**. Therefore, the variable user will be _**Daniel's**_ profile. To fix this, I would advise you to use `instance variables (@user1, @user2)` to store the user profiles and use them where needed. 👍 💻 

![image](https://github.com/Kaghenimbale/blog-app/assets/77106962/710950e8-691f-496e-8b16-795e57908860)